compiler_gym/service/BUILD

```diff
@@ -12,6 +12,8 @@ py_library(
     deps = [
         ":compilation_session",
         ":connection",
+        # TODO: add this after circular dependencies are resolved
+        # ":client_service_compiler_env",
```

Comment:
I wonder if the easies thing here would be to merge `:service` and `:client_service_compiler_env`, so targets need only depend on `//compiler_gym/service`?

Reply:
(same for cmake)

Reply:
This will cause circular dependency. I have more elaborate explanation in another comment.

Reply:
Okay, in that case could you update the TODO note to reference this?

```
TODO(github.com/facebookresearch/CompilerGym/pull/633): ...
```